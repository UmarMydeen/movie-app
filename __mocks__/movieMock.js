const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster',
    imdbRating: '9.2',
    Poster: 'mockUrl'
}

export default movieMock;