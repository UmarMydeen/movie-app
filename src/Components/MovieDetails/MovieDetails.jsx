import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import getMovideById from '../../actions/getMovieById';

function MovieDetails({
    match: { params: { id } },
    actionGetMovideById
}) {
    const [movie, setMovie] = React.useState({});

    React.useEffect(() => {
        const response = actionGetMovideById(id);
        response.then(data => (setMovie(data)));
    }, [])

    const { Stills, Title, Plot, imdbRating, Poster } = movie;

    return (
        <div className="container">
            <div className="row row-cols-6 row-cols-md-5">
                <div className="col-md-7">
                    <dl className="row">
                        <dt className="col-sm-3">Title</dt>
                        <dd className="col-sm-9" data-testid="titles">{Title}</dd>

                        <dt className="col-sm-3">Rating</dt>
                        <dd className="col-sm-9">{imdbRating}</dd>

                        <dt className="col-sm-3">Plot</dt>
                        <dd className="col-sm-9">{Plot}</dd>
                    </dl>
                </div>
                <div className="col-md-5">
                    <div className="row">
                        <img src={Poster} className="d-block w-50" alt="..." />
                    </div>
                    <div className="row">
                        {
                            Stills && Stills.map(stillPath => (
                                <div className="col-3 m-1">
                                    <img src={stillPath} className="d-block w-100" alt="..." />
                                </div>
                            ))
                        }
                    </div>
                </div>
            </div >
        </div>
    );
}

MovieDetails.propTypes = {
    match: {
        params: {
            id: PropTypes.string.isRequired
        }
    },
    actionGetMovideById: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => {
    return {
        actionGetMovideById: (id) => dispatch(getMovideById(id))
    }
};

export { MovieDetails as TestableMovieDetails };

const hocChain = compose(
    connect(null, mapDispatchToProps),
);

export default hocChain(MovieDetails);