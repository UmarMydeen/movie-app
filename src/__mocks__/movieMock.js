const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster'
}

export default movieMock;