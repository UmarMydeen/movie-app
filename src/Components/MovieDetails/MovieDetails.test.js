import { getByTestId, render, screen } from '@testing-library/react';
import { TestableMovieDetails as MovieDetails } from './MovieDetails';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from '../../Reducers/index';
import movieMock from '../../../__mocks__/movieMock';
import { act } from "react-dom/test-utils";

let mockProps;

const initialState = {
    movies: [],
    filters: {},
    sort: {}
};

const store = createStore(rootReducer, initialState);
const mockPromiseResponse = (id) => (new Promise(resolve => (resolve(movieMock))))

beforeEach(() => {
    mockProps = {
        match: {
            params: {
                id: 'MockId'
            }
        },
        actionGetMovideById: mockPromiseResponse
    }
})

test('should render component correctly', () => {
    const component = render(
        <Provider store={store}>
            <MovieDetails {...mockProps} />
        </Provider>
    );
    expect(component).toMatchSnapshot();
});