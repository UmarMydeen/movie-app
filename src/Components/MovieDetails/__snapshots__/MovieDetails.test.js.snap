// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="row row-cols-6 row-cols-md-5"
        >
          <div
            class="col-md-7"
          >
            <dl
              class="row"
            >
              <dt
                class="col-sm-3"
              >
                Title
              </dt>
              <p
                data-testid="title"
              />
              <dd
                class="col-sm-9"
                data-testid="titles"
              />
              <dt
                class="col-sm-3"
              >
                Rating
              </dt>
              <dd
                class="col-sm-9"
              />
              <dt
                class="col-sm-3"
              >
                Plot
              </dt>
              <dd
                class="col-sm-9"
              />
            </dl>
          </div>
          <div
            class="col-md-5"
          >
            <div
              class="row"
            >
              <img
                alt="..."
                class="d-block w-50"
              />
            </div>
            <div
              class="row"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="row row-cols-6 row-cols-md-5"
      >
        <div
          class="col-md-7"
        >
          <dl
            class="row"
          >
            <dt
              class="col-sm-3"
            >
              Title
            </dt>
            <p
              data-testid="title"
            />
            <dd
              class="col-sm-9"
              data-testid="titles"
            />
            <dt
              class="col-sm-3"
            >
              Rating
            </dt>
            <dd
              class="col-sm-9"
            />
            <dt
              class="col-sm-3"
            >
              Plot
            </dt>
            <dd
              class="col-sm-9"
            />
          </dl>
        </div>
        <div
          class="col-md-5"
        >
          <div
            class="row"
          >
            <img
              alt="..."
              class="d-block w-50"
            />
          </div>
          <div
            class="row"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
